core/templates/pages/exploration-editor-page/services/router.service.ts

```diff
@@ -165,11 +165,21 @@ angular.module('oppia').factory('RouterService', [
     };
 
     var _savePendingChanges = function() {
-      // Sometimes, AngularJS throws a "Cannot read property $$nextSibling of
-      // null" error. To get around this we must use $timeout().
-      $timeout(function() {
+      try {
```

Comment:
I'm concerned about why you needed to change this. Wouldn't using done() in the test have been sufficient?

Reply:
The `try/catch` was being used before `$timeout`. I changed to `$timeout` because I didn't find a solution for testing `try/catch` before. But `$timeout` turns the test very tricky and unstable. So `try/catch` is the best approach for it.
I've tried to use the code inside `catch` directly as well but then e2e were failing.

Reply:
OK. Could you please point to the PR which changed this to $timeout? I just need to verify (as a reviewer) that these changes are safe. Thanks!

Reply:
It's #8696